solana/programs/example-native-token-transfers/src/instructions/admin.rs

```diff
@@ -154,6 +156,167 @@ pub fn claim_ownership(ctx: Context<ClaimOwnership>) -> Result<()> {
     )
 }
 
+// * Set token authority
+
+#[derive(Accounts)]
+pub struct SetTokenAuthority<'info> {
+    #[account(
+        has_one = owner,
+        constraint = config.paused @ NTTError::NotPaused,
+    )]
+    pub config: Account<'info, Config>,
+
+    pub owner: Signer<'info>,
+
+    #[account(
+        mut,
+        address = config.mint,
+    )]
+    /// CHECK: the mint address matches the config
+    pub mint: InterfaceAccount<'info, token_interface::Mint>,
+
+    #[account(
+        seeds = [crate::TOKEN_AUTHORITY_SEED],
+        bump,
+        address = mint.mint_authority.unwrap() @ NTTError::InvalidMintAuthority
```

Comment:
I'd either get rid of this check (since it is ultimately unnecessary, and I don't like diluting code with unnecessary checks thus making it unclear what is or isn't critical), or at least also have it in the revert version so it is at least consistent.

Reply:
This is not unnecessary with regards to the `set_token_authority` (checked) ix as we init the `pending_token_authority` without doing any transfer so we need to ensure the mint authority is valid.
It is unnecessary wrt the `set_token_authority_one_step_unchecked` though as the token program will check itself.

I can remove this constraint from `SetTokenAuthority` and add it in the `SetTokenAuthorityChecked` struct to remove redundant checks like so:
```rust
#[derive(Accounts)]
pub struct SetTokenAuthorityChecked<'info> {
#[account(
    constraint = common.token_authority.key() == common.mint.mint_authority.unwrap() @ NTTError::InvalidMintAuthority
)]
pub common: SetTokenAuthority<'info>,
...
}
```

Reply:
Yeah, good point.